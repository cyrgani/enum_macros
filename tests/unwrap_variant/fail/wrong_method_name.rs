use enum_macros::UnwrapVariant;

#[derive(UnwrapVariant)]
enum Test {
    #[unwrap(mut)]
    A(String),
    #[unwrap(ref)]
    B(usize),
}

fn main() {
    let _a: &String = Test::unwrap_A_ref();
    let _b: &mut usize = Test::unwrap_B_mut();
}
