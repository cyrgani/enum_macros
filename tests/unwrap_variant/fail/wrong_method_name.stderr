error[E0599]: no variant or associated item named `unwrap_A_ref` found for enum `Test` in the current scope
  --> tests/unwrap_variant/fail/wrong_method_name.rs:12:29
   |
4  | enum Test {
   | --------- variant or associated item `unwrap_A_ref` not found for this enum
...
12 |     let _a: &String = Test::unwrap_A_ref();
   |                             ^^^^^^^^^^^^ variant or associated item not found in `Test`
   |
help: there is a method `unwrap_b_ref` with a similar name
   |
12 |     let _a: &String = Test::unwrap_b_ref();
   |                             ~~~~~~~~~~~~

error[E0599]: no variant or associated item named `unwrap_B_mut` found for enum `Test` in the current scope
  --> tests/unwrap_variant/fail/wrong_method_name.rs:13:32
   |
4  | enum Test {
   | --------- variant or associated item `unwrap_B_mut` not found for this enum
...
13 |     let _b: &mut usize = Test::unwrap_B_mut();
   |                                ^^^^^^^^^^^^ variant or associated item not found in `Test`
   |
help: there is a method `unwrap_a_mut` with a similar name
   |
13 |     let _b: &mut usize = Test::unwrap_a_mut();
   |                                ~~~~~~~~~~~~
